src/compiler/crystal/semantic/main_visitor.cr

```diff
@@ -93,7 +93,7 @@ module Crystal
     @in_type_args = 0
 
     @while_stack : Array(While)
-    @type_filters : TypeFilters?
+    @type_filters : TypeFilters = TypeFilters.new
```

Comment:
Is there a way to avoid creating one of these each time? I see all of the `nil` were replaced with these and this will make the compiler consume more memory.